common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -7136,6 +7156,11 @@ public boolean isWebUiQueryInfoCacheEnabled() {
     return isWebUiEnabled() && this.getIntVar(ConfVars.HIVE_SERVER2_WEBUI_MAX_HISTORIC_QUERIES) > 0;
   }
 
+  public boolean isQueryHistoryExplainEnabled() {
```

Comment:
the method is used just in `Compiler#explainOutput` + it reads as explain history not 'explain plan'. I would inline and use config directly

Reply:
okay, I'm doing this in Compiler method
```
    boolean queryHistoryExplainPlanEnabled = conf.getBoolVar(
        HiveConf.ConfVars.HIVE_QUERY_HISTORY_ENABLED) && conf.getBoolVar(
        HiveConf.ConfVars.HIVE_QUERY_HISTORY_EXPLAIN_PLAN_ENABLED);
```